src/pymanifold.py

```diff
@@ -125,53 +125,56 @@ def channel(self,
             self.dg.edges[port_from, port_to][key] = attr
         return
 
-    def retrieve(port_in, port_out, attr):
-      if(port_out == "None"):
-        return self.dg.edges[port_in][attr]
+    def retrieve(port_in, attr, port_out=None):
```

Comment:
Since you have port_out=None in each of the get methods its not needed here, otherwise within each of the get methods you will have to call ```self.retrieve(port_in, attr, port_out=port_out``` since in Python you need to specify the name of which variable you're providing when it has a default value like None. This is because there can be many of these parameters with default values that don't need to be provided when calling that function, even though in our case there only one it is still required that you specify.